caikit/core/toolkit/reversible_hasher.py

```diff
@@ -0,0 +1,67 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""The ReversibleHasher provides a simple obfuscation tool for munging strings
+in a reversible way. It is useful when needing a repeatable way to obfuscate a
+string that may come from a configuration detail that should not be
+transparently visible to a user, but needs to be trivially reversed, possibly
+with entirely new in-memory state.
+
+WARNING: This utility is intentionally not a cryptographically secure hash! It
+    is explicitly designed to be reversible.
+"""
+
+# Standard
+import random
+import string
+
+
+class _ReversibleHasherMeta(type):
+    """The _ReversibleHasherMeta is used here to ensure that the random state is
+    only temporarily seeded at import time.
+
+    WARNING: If this class is dynamically imported in a multi-threaded
+        application, other threads that share the random state will be effected
+        by the temporary seed!
```

Comment:
Is this because of setting the seed via `random.seed` ? Since quite a bit of the applications rely on random number, like in NLP, having a different random seed can have unprecedented side-effects.

Wondering if this can be resolved by using `random.Random` and essentially doing the random-number generation for this class in a more localized manner?

Reply:
🤦🤦🤦 Yep, `random.Random` is a MUCH better solution. I was doing all of this metadata junk to try to accomplish what `random.Random` is explicitly designed to do. Will refactor accordingly